core/component.go

```diff
@@ -118,6 +118,18 @@ func NewComponent(ctx log.Interface, serviceName string, announcedAddress string
 		go http.ListenAndServe(fmt.Sprintf(":%d", healthPort), nil)
 	}
 
+	if monitors := viper.GetStringMapString("monitor-servers"); len(monitors) != 0 {
+		component.Monitors = map[string]pb_noc.MonitorClient{}
```

Comment:
so what do you want me to do here? I think checking for `nil` goes well together with checking if `gateway.Monitor` is `nil`